app/views/taxonomies/_form.html.erb

```diff
@@ -62,122 +62,24 @@
-            <%= multiple_selects f, :realms, Realm, taxonomy.selected_or_inherited_ids[:realm_ids],
-                                {:disabled => taxonomy.used_or_inherited_ids[:realm_ids], :label => _('Select realms')},
-                                {'data-mismatches' => taxonomy.need_to_be_selected_ids[:realm_ids].to_json,
-                                 'data-inheriteds' => taxonomy.inherited_ids[:realm_ids].to_json,
-                                 'data-useds' => taxonomy.used_ids[:realm_ids].to_json } %>
-          </div>
-        <% end %>
-     <% end %>
-
-     <% if User.current.allowed_to?(:view_environments) %>
-        <div class="tab-pane" id="environments">
-          <%= checkbox_f(f, :ignore_types, {:label => _("All environments"), :multiple => true, :onchange => 'ignore_checked(this)'}, "Environment") %>
-          <%= multiple_selects f, :environments, Environment, taxonomy.selected_or_inherited_ids[:environment_ids],
-                               {:disabled => taxonomy.used_and_selected_or_inherited_ids[:environment_ids], :label => _('Select environments')},
-                               {'data-mismatches' => taxonomy.need_to_be_selected_ids[:environment_ids].to_json,
-                                'data-inheriteds' => taxonomy.inherited_ids[:environment_ids].to_json,
-                                'data-useds' => taxonomy.used_ids[:environment_ids].to_json } %>
-        </div>
+        <%= show_resource_if_allowed(f, taxonomy, "subnets") %>
+        <%= show_resource_if_allowed(f, taxonomy, "compute_resources") %>
+        <%= show_resource_if_allowed(f, taxonomy, "media") %>
+        <%= show_resource_if_allowed(f, taxonomy, "templates") %>
+        <%= show_resource_if_allowed(f, taxonomy, "domains") %>
+        <%= show_resource_if_allowed(f, taxonomy, "realms") %>
       <% end %>
 
-      <% if User.current.allowed_to?(:view_hostgroups) %>
-        <div class="tab-pane" id="hostgroups">
-          <%= checkbox_f(f, :ignore_types, {:label => _("All host groups"), :multiple => true, :onchange => 'ignore_checked(this)'}, "Hostgroup") %>
-          <%= multiple_selects f, :hostgroups, Hostgroup, taxonomy.selected_or_inherited_ids[:hostgroup_ids],
-                               {:disabled => taxonomy.used_and_selected_or_inherited_ids[:hostgroup_ids], :label => _('Select host groups')},
-                               {'data-mismatches' => taxonomy.need_to_be_selected_ids[:hostgroup_ids].to_json,
-                                'data-inheriteds' => taxonomy.inherited_ids[:hostgroup_ids].to_json,
-                                'data-useds' => taxonomy.used_ids[:hostgroup_ids].to_json } %>
-        </div>
-      <% end %>
+      <%= show_resource_if_allowed(f, taxonomy, "environments") %>
+      <%= show_resource_if_allowed(f, taxonomy, "hostgroups") %>
 
       <% if taxonomy.is_a?(Location) && SETTINGS[:locations_enabled] %>
```

Comment:
Combine the conditions, and use plain `if` instead of `elseif`.
``` HTML+ERB
<% if taxonomy.is_a?(Location) && SETTINGS[:locations_enabled] && User.current.allowed_to?(:view_organizations)  %>
```